### PR TITLE
restore no_std after diman_lib was split out

### PR DESCRIPTION
**Pull request by jedbrown**
When we added no_std (https://github.com/Tehforsch/diman/pull/60), crates/diman_lib did not yet exist. It's now missing the no_std annotation, which has allowed tests to pass (since the tests are all running for architectures that support std). Rust does not prevent no_std crates from calling crates that use std (this is confusing).

Adding no_std to diman_lib moves the dependency on num-traits into diman_lib and we drop the ratio module when we don't have either std or num-traits-libm (since we need powf for it to work).

I've tested this with `--target=nvptx64-nvidia-cuda`.

### CODE REVIEW

File: crates/diman_lib/src/magnitude.rs [hunk 1]
```diff
@@ -3,6 +3,8 @@ use core::{
     ops::{Div, Mul},
 };
 
+use crate::num_traits_reexport::*; // either Float or FloatCore
```
Comment (jedbrown) [inline crates/diman_lib/src/magnitude.rs:6]: This is a stylistic choice that I have no attachment to. We could instead use `Float` or `FloatCore` depending on cfg.

File: crates/diman_lib/Cargo.toml [hunk 2]
```diff
@@ -9,4 +9,9 @@ description = "Commonly used types for internal use in diman."
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/tehforsch/diman"
 
+[features]
+std = []
+num-traits-libm = ["num-traits/libm"]
```
Comment (jedbrown) [inline crates/diman_lib/Cargo.toml:14]: I chose to make neither default. If you envision external users of `diman_lib`, perhaps we should have `default = ["std"]`.

Comment (Tehforsch) [reply crates/diman_lib/Cargo.toml:14]: I don't really see anyone using diman_lib outside of diman, so I think your choice is fine.

File: crates/diman_lib/src/magnitude.rs (hunk 1 above)
Comment (Tehforsch) [reply crates/diman_lib/src/magnitude.rs:6]: I think this needs to be 
```rust
#[cfg(feature = "num-traits-libm")]
pub use num_traits::float::Float;

#[cfg(not(feature = "num-traits-libm"))]
pub use num_traits::float::FloatCore;
```

just to get rid of the warning.

**Review by Tehforsch — CHANGES_REQUESTED** [review]
Ah, not testing no-std after diman_lib was my bad, sorry about that. I didn't realize that one could just use std in dependencies even if the crate is `no_std`. 

Thanks for the fix - from my side this looks good, except for the warning because of the glob import.

**Comment by Tehforsch** [comment]
I've fixed a few minor problems that came up and added a no_std build on `nvptx64-nvidia-cuda` to the CI, to hopefully avoid problems like this in the future. Could you just quickly confirm that the new version of this PR works for you? If so I'll merge it and release `0.5.1` containing this and a few other small fixes.

**Review by Tehforsch — APPROVED** [review]

File: crates/diman_lib/src/dimension_exponent.rs [hunk 3]
```diff
@@ -2,6 +2,13 @@ use core::ops::{AddAssign, Mul, Neg};
 
 use crate::magnitude::Magnitude;
 
+// For some reason, this use statement is never recognized as used
+// even when I build the crates with no std, where removing the use
+// statement means that powi below cannot be used.
+#[allow(unused)]
+#[cfg(not(any(feature = "std")))]
+use num_traits::float::FloatCore;
```
Comment (jedbrown) [inline crates/diman_lib/src/dimension_exponent.rs:10]: A minimized version of this issue: https://github.com/rust-lang/rust/issues/127852#issuecomment-2278496055

**Review by jedbrown — COMMENTED** [review]
Looks good, thanks. And sorry about the issues in the first version and taking a while to circle back.

**Comment by Tehforsch** [comment]
> Looks good, thanks. And sorry about the issues in the first version and taking a while to circle back.

No worries!

**Pull request merged**